addons/account/models/account_payment.py

```diff
@@ -54,7 +54,12 @@ class account_payment(models.Model):
 
     state = fields.Selection([('draft', 'Draft'), ('posted', 'Validated'), ('sent', 'Sent'), ('reconciled', 'Reconciled'), ('cancelled', 'Cancelled')], readonly=True, default='draft', copy=False, string="Status")
     payment_type = fields.Selection([('outbound', 'Send Money'), ('inbound', 'Receive Money'), ('transfer', 'Internal Transfer')], string='Payment Type', required=True, readonly=True, states={'draft': [('readonly', False)]})
+    _payment_methods = fields.Many2many('account.payment.method', compute='_compute_payment_methods')
     payment_method_id = fields.Many2one('account.payment.method', string='Payment Method', required=True, readonly=True, states={'draft': [('readonly', False)]},
+        domain="""[
+            ('payment_type', '=', ((payment_type == 'inbound') and 'inbound' or 'outbound')),
```

Comment:
Is there any reason not to fold this into _payment_methods?

Reply:
(I guess that _payment_methods is used in other places than here though even then shouldn't the payment type match? It doesn't in the original but that's still a bit weird)